spmd/tensor/device_mesh.py

```diff
@@ -391,3 +392,22 @@ def reduce_scatter_base(
 
             chunks = reduced_tensor.chunk(num_chunks, dim=shard_dim)
             return self.scatter(chunks, mesh_dim=mesh_dim)
+
+    def all_to_all(
+        self, input_tensor_list: List[torch.Tensor], mesh_dim: int = 0
+    ) -> List[torch.Tensor]:
+        # TODO: shall we make input tensors contiguous???
```

Comment:
we should always make collective inputs be contiguous tensors, otherwise the collective won't work correctly as we expected.

Reply:
Thx for explanation!